### ISSUE TITLE
is TableMap.Columns guaranteed to be in the same order as the corresponding struct's fields?

### ISSUE THREAD

**Issue by bigblind** [go-gorp/gorp#239]
In other words, will `tableMap.Columns[0]` return the ColumnMap associated with the field returned by `reflect.TypeOf(ModelStruct).Field(0)`? If so, can I rely on it, or is it concidered an implementation detail?

**Comment by nelsam** [comment]
Not necessarily.  TableMap.Columns is a slice of all detected columns - typically in the order you would expect, but you shouldn't think of it as a 1-to-1 mapping.  For example:

```go
type Foo struct {
    Bacon string // this is tableMap.Colums[0]
    Eggs int64 // this is tableMap.Columns[1]
}

type Bar struct {
    Foo // this is reflect.TypeOf(Bar{}).Field(0)
    Toast float32 // this is tableMap.Columns[2] and reflect.TypeOf(Bar{}).Field(1)
}
```

I'm undecided on whether or not you should rely on the column order, so I guess (at least for now) I would say you should consider it an implementation detail.  @GeertJohan do you have any thoughts?

**Comment by GeertJohan** [comment]
I agree with @nelsam. This is an implementation detail and might change. We could spec this though.. If we do, it might be best to add a getter method that takes an int and returns the field. It needs to be thought over though, because I wonder what the use case for this would be and it looks like it is adding to API bloat.
